### PR TITLE
feat!: add max_time_travel_hours attribute

### PR DESCRIPTION
**Pull request by kpocius**
The attribute was introduced in the provider https://github.com/hashicorp/terraform-provider-google/pull/12830 but I can't find it referenced in a changelog, so not sure which version. It is, however, causing unintended drift.

Set the default to 168 hours, which is the default when creating a dataset via UI.

### CODE REVIEW

**Comment by comment-bot-dev** [comment]
@kpocius
Thanks for the PR! 🚀
Unfortunately it looks like some of our CI checks failed. See the [Contributing Guide](../blob/master/CONTRIBUTING.md) for details.
- ⚠️check_documentation
 The documentation needs to be regenerated. Please run `make generate_docs`. 
 ```bash 
Checking for documentation generation
diff -r '--exclude=.terraform' '--exclude=.kitchen' '--exclude=autogen' '--exclude=*.tfvars' '--exclude=*metadata.yaml' /workspace/README.md /tmp/tmp.YSUIUrcV7g/generate_docs/workspace/README.md
191d190
< | max_time\_travel\_hours | Defines the time travel window in hours | `number` | `168` | no |
198a198
> | max\_time\_travel\_hours | Defines the time travel window in hours | `number` | `168` | no |
Error: Documentation generation has not been run, please run the
'make docker_generate_docs' command and commit the above changes.
 ```

**Comment by comment-bot-dev** [comment]
@kpocius
Thanks for the PR! 🚀
Unfortunately it looks like some of our CI checks failed. See the [Contributing Guide](../blob/master/CONTRIBUTING.md) for details.
- ⚠️check_documentation
 The documentation needs to be regenerated. Please run `make generate_docs`. 
 ```bash 
Checking for documentation generation
diff -r '--exclude=.terraform' '--exclude=.kitchen' '--exclude=autogen' '--exclude=*.tfvars' '--exclude=*metadata.yaml' /workspace/README.md /tmp/tmp.wjzZ6Ttacd/generate_docs/workspace/README.md
191d190
< | max_time\_travel\_hours | Defines the time travel window in hours | `number` | `168` | no |
198a198
> | max\_time\_travel\_hours | Defines the time travel window in hours | `number` | `168` | no |
Error: Documentation generation has not been run, please run the
'make docker_generate_docs' command and commit the above changes.
 ```

**Comment by comment-bot-dev** [comment]
@kpocius
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

File: variables.tf [hunk 1]
```diff
@@ -55,6 +55,12 @@ variable "default_table_expiration_ms" {
   default     = null
 }
 
+variable "max_time_travel_hours" {
+  description = "Defines the time travel window in hours"
+  type        = number
+  default     = 168
```
Comment (bharathkkb) [inline variables.tf:61]: Let's default to null so we can re use the provider default

```suggestion
  default     = null
```

**Review by bharathkkb — COMMENTED** [review]
Thanks for the PR @kpocius 
Looks like this was added in [4.42 to the provider](https://github.com/hashicorp/terraform-provider-google/blob/main/CHANGELOG.md#4420-october-31-2022) and we can include this in our upcoming major release. Can you bump this here?
https://github.com/terraform-google-modules/terraform-google-bigquery/blob/d4f61d3ee2427d8d42cab767c0326074c56d2c17/versions.tf#L23

**Comment by kpocius** [comment]
@bharathkkb, I made the requested changed. Can you please re-run CI/CD pipeline?

**Review by bharathkkb — APPROVED** [review]

**Pull request merged**